libraries/matrixtest/src/main/kotlin/io/element/android/libraries/matrixtest/room/InMemoryRoomSummaryDataSource.kt

```diff
@@ -23,9 +23,19 @@ import kotlinx.coroutines.flow.StateFlow
 
 class InMemoryRoomSummaryDataSource : RoomSummaryDataSource {
 
+    private val roomSummariesFlow = MutableStateFlow<List<RoomSummary>>(emptyList())
+
+    suspend fun postRoomSummary(roomSummaries: List<RoomSummary>) {
+        roomSummariesFlow.emit(roomSummaries)
+    }
+
     override fun roomSummaries(): StateFlow<List<RoomSummary>> {
-        return MutableStateFlow(emptyList())
+        return roomSummariesFlow
     }
 
-    override fun setSlidingSyncRange(range: IntRange) = Unit
+    var latestSlidingSyncRange: IntRange? = null
+
+    override fun setSlidingSyncRange(range: IntRange) {
```

Comment:
Maybe reflect the change on the roomSummariesFlow? I'd like to see the data being changed too

Reply:
to my understanding, this can be un-correlated for unit test. if we change the `roomSummariesFlow` we are giving too much intelligence to the `FakeRoomSummaryDataSource` and this will not be a unit test.